src/main/java/com/iexec/core/task/TaskController.java

```diff
@@ -116,9 +154,27 @@ public ResponseEntity<TaskLogs> getTaskLogs(@PathVariable("chainTaskId") String
     })
     public ResponseEntity<ComputeLogs> getComputeLogs(
             @PathVariable("chainTaskId") String chainTaskId,
-            @PathVariable("walletAddress") String walletAddress) {
+            @PathVariable("walletAddress") String walletAddress,
+            @RequestHeader("Authorization") String bearerToken) {
+        String computeLogsRequester = challengeService.getWalletAddressFromBearerToken(bearerToken);
+        if (!challengeService.isValidToken(bearerToken)
+            || !isTaskRequester(computeLogsRequester, chainTaskId)) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
         return taskLogsService.getComputeLogs(chainTaskId, walletAddress)
                 .map(ResponseEntity::ok)
                 .orElse(ResponseEntity.notFound().build());
     }
+
+    /**
+     * Checks if requester address from bearer token is the same as the address used to buy the task execution.
+     * @param logsRequester Wallet address of requester asking task or compute logs
+     * @param chainTaskId Task for which outputs are requested
+     * @return true if the user requesting computation outputs was the one to buy the task, false otherwise
+     */
+    private boolean isTaskRequester(String logsRequester, String chainTaskId) {
+        TaskDescription taskDescription = iexecHubService.getTaskDescription(chainTaskId);
+        return taskDescription != null && taskDescription.getRequester().equalsIgnoreCase(logsRequester);
```

Comment:
Could `getRequester()` be null?

Reply:
I will invert to `logsRequester.equalsIgnoreCase(taskDescription.getRequester())`.
This should be safer

Reply:
inversion done in 7f43639